sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avro_serializer/_avro_serializer.py

```diff
@@ -0,0 +1,114 @@
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import abc
+from typing import BinaryIO, Union, Type, TypeVar, Optional, Any, Dict
+from io import BytesIO
+import avro
+from avro.io import DatumWriter, DatumReader, BinaryDecoder, BinaryEncoder
+
+try:
+    ABC = abc.ABC
+except AttributeError:  # Python 2.7, abc exists, but not ABC
+    ABC = abc.ABCMeta("ABC", (object,), {"__slots__": ()})  # type: ignore
+
+ObjectType = TypeVar("ObjectType")
+
+
+class AvroObjectSerializer(ABC):
+
+    def __init__(self, codec=None):
+        """A Avro serializer using avro lib from Apache.
+        :param str codec: The writer codec. If None, let the avro library decides.
+        """
+        self._writer_codec = codec
+        self._schema_writer_cache = {}  # type: Dict[str, DatumWriter]
+        self._schema_reader_cache = {}  # type: Dict[str, DatumReader]
+
+    def serialize(
+        self,
+        data,  # type: ObjectType
+        schema,  # type: Optional[Any]
+    ):
+        # type: (...) -> bytes
+        """Convert the provided value to it's binary representation and write it to the stream.
+        Schema must be a Avro RecordSchema:
+        https://avro.apache.org/docs/1.10.0/gettingstartedpython.html#Defining+a+schema
+        :param data: An object to serialize
+        :param schema: A Avro RecordSchema
+        """
+        if not schema:
+            raise ValueError("Schema is required in Avro serializer.")
+
+        if not isinstance(schema, avro.schema.Schema):
+            schema = avro.schema.parse(schema)
+
+        try:
+            writer = self._schema_writer_cache[str(schema)]
+        except KeyError:
+            writer = DatumWriter(schema)
+            self._schema_writer_cache[str(schema)] = writer
+
+        stream = BytesIO()
+
+        writer.write(data, BinaryEncoder(stream))
+        encoded_data = stream.getvalue()
+
+        stream.close()
+        return encoded_data
+
+    def deserialize(
+        self,
+        data,  # type: Union[bytes, BinaryIO]
+        schema,  # type:
+        return_type=None,  # type: Optional[Type[ObjectType]]  # pylint: disable=unused-argument
```

Comment:
Let's remove it for now since we don't use it nor inherit from azure-core anything

Reply:
removed